### PR TITLE
(maint) add hup helper

### PR DESCRIPTION
**Pull request by erikPrime**

### CODE REVIEW

File: acceptance/lib/helper.rb [hunk 1]
```diff
@@ -259,7 +285,34 @@ def https_request(url, request_method, cert, key, body = nil)
     response = http.request(request)
   end
 
-
+  def hup_server(host = master, timeout = 30)
+    pid=on(host, 'cat /var/run/puppetlabs/puppetserver/puppetserver').stdout.chomp
```
Comment (kurtwall) [inline acceptance/lib/helper.rb]: Not really excited about a hard-coded path. Too bad we don't have a `puppetserver config print pidfile`. The path seems unlikely to change, but...

Comment (erikPrime) [inline acceptance/lib/helper.rb]: I couldn't figure out what to query in order to make this a variable.

File: acceptance/lib/helper.rb [hunk 2]
```diff
@@ -259,7 +285,34 @@ def https_request(url, request_method, cert, key, body = nil)
...
+    on(host, "kill -HUP #{pid}")
+    url="https://#{host}:8140/puppet/v3/status"
+    cert=get_cert(master)
+    key=get_key(master)
+    response_code = "0"
+    sleeptime = 1
+    while response_code != "404" && timeout > 0
+      sleep sleeptime
+      begin
+        response = https_request(url, 'GET', cert, key)
```
Comment (kurtwall) [inline acceptance/lib/helper.rb]: Not sure it matters here, but I coded `https_request` to use keywords for the method parameter, thus:
`https_request(url, :get, cert, key)`

File: acceptance/lib/helper.rb [hunk 3]
```diff
@@ -259,7 +285,34 @@ def https_request(url, request_method, cert, key, body = nil)
...
+      rescue StandardError => e
+        expected_errors = [ Errno::ECONNREFUSED, Errno::ECONNRESET, 
+                            OpenSSL::SSL::SSLError ]
+        if expected_errors.include?e.class 
+          raise e
+        else
+          # Does this message violate the Wolfe Principle?
```
Comment (kurtwall) [inline acceptance/lib/helper.rb]: "Wolfe Principle"?

Comment (erikPrime) [inline acceptance/lib/helper.rb]: Wolfe Principle = If it isn't an error, or if it isn't broken, don't log it.

File: acceptance/lib/helper.rb (hunk 1 above)
Comment (rlinehan) [inline acceptance/lib/helper.rb]: you could do `puppet master --configprint rundir` + `puppetserver/puppetserver` I think.

**Comment by kurtwall** [comment]
If you'd put whitespace around operators consistently, I would shake your hand.

File: acceptance/lib/helper.rb (hunk 2 above)
Comment (erikPrime) [inline acceptance/lib/helper.rb]: ? Not in the copy I'm looking at?  Do you mean in pe-puppetserver-extensions?  Or in a later commit?

**Comment by erikPrime** [comment]
White space added...

**Comment by kurtwall** [comment]
🙈

**Pull request merged**